test/tracing/test_tracer.py

```diff
@@ -61,3 +73,57 @@ def test_disable_tracing(self) -> None:
         disable_tracing()
         assert isinstance(tracer.actual_tracer, NullTracer)
         assert is_tracing_enabled() is False
+
+
+class TestAutoEnableTracer:
+    @pytest.fixture()
+    def configured_opentelemetry_tracing(self) -> None:
+        resource = Resource(attributes={SERVICE_NAME: "haystack-testing"})
+
+        traceProvider = TracerProvider(resource=resource)
+        processor = SimpleSpanProcessor(InMemorySpanExporter())
+        traceProvider.add_span_processor(processor)
+
+        # We can't uset `set_tracer_provider` here, because opentelemetry has a lock to only set it once
+        opentelemetry.trace._TRACER_PROVIDER = traceProvider
+
+        yield
+
+        # unfortunately, there's no cleaner way to reset the global tracer provider
+        opentelemetry.trace._TRACER_PROVIDER = None
+        disable_tracing()
+
+    def test_skip_auto_enable_tracer_if_already_configured(self) -> None:
+        my_tracker = Mock(spec=Tracer)  # anything else than `NullTracer` works for this test
+        enable_tracing(my_tracker)
+
+        auto_enable_tracing()
+
+        assert tracer.actual_tracer is my_tracker
+
+    def test_skip_auto_enable_if_tracing_disabled_via_env(
+        self, monkeypatch: MonkeyPatch, configured_opentelemetry_tracing: None
```

Comment:
that's the typing on `monkeypatch` @masci

Reply:
The import from a `_` module biased me 😅 and I misread the calling code, thanks for pointing this out!